delira/io/tf.py

```diff
@@ -35,3 +37,55 @@ def load_checkpoint(file: str, model=None):
     # this for memory leak
     tf.train.Saver().restore(model._sess, file)
     return {}
+
+
+def _create_varlist(model: AbstractTfEagerNetwork = None,
+                    optimizer: typing.Dict[str, tf.train.Optimizer] = None):
+    variable_list = []
+
+    if model is not None:
+        variable_list += model.variables
```

Comment:
This raises an actual error instead of a warning due to tf internals if model has not been built or called once.

ValueError: Weights for model sequential have not yet been created. Weights are created when the Model is first called on inputs or `build()` is called with an `input_shape`.

Reply:
Yeah, but I think it's nicer this way since we don't have to raise an Exception here this way (and this should work in Epochs other than 0)